pkg/controller/catalog/storage/mem_storage.go

```diff
@@ -16,211 +16,45 @@ limitations under the License.
 
 package storage
 
-import (
-	"fmt"
-	"log"
-	"strings"
-
-	model "github.com/kubernetes-incubator/service-catalog/model/service_controller"
-)
-
-type bindingPair struct {
-	binding    *model.ServiceBinding
-	credential *model.Credential
-}
-
 type memStorage struct {
-	brokers map[string]*model.ServiceBroker
-	// This gets fetched when a SB is created (or possibly later when refetched).
-	// It's static for now to keep compatibility, seems like this could be more dynamic.
-	catalogs map[string]*model.Catalog
-	// maps instance ID to instance
-	services map[string]*model.ServiceInstance
-	// maps binding ID to binding
-	// TODO: support looking up all bindings for a service instance.
-	bindings map[string]*bindingPair
+	brokers        *memStorageBroker
+	serviceClasses *memStorageServiceClass
+	instances      map[string]*memStorageInstance
+	bindings       map[string]*memStorageBinding
 }
 
 // CreateMemStorage creates an instance of Storage interface, backed by memory.
 func CreateMemStorage() Storage {
 	return &memStorage{
-		brokers:  make(map[string]*model.ServiceBroker),
-		catalogs: make(map[string]*model.Catalog),
-		services: make(map[string]*model.ServiceInstance),
-		bindings: make(map[string]*bindingPair),
+		brokers:        newMemStorageBroker(),
+		serviceClasses: newMemStorageServiceClass(),
+		instances:      make(map[string]*memStorageInstance),
+		bindings:       make(map[string]*memStorageBinding),
 	}
 }
 
-func (s *memStorage) GetInventory() (*model.Catalog, error) {
-	services := []*model.Service{}
-	for _, v := range s.catalogs {
-		services = append(services, v.Services...)
-	}
-	return &model.Catalog{Services: services}, nil
-}
-
-func (s *memStorage) ListBrokers() ([]*model.ServiceBroker, error) {
-	b := []*model.ServiceBroker{}
-	for _, v := range s.brokers {
-		b = append(b, v)
-	}
-	return b, nil
+func (m *memStorage) Brokers() BrokerStorage {
```

Comment:
yes, this is k8s convention, except for the name `BrokerStorage` and the other `*Storage` types below, which will be changed later when we generate a client. See [`CoreV1Client`](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/clientset_generated/clientset/typed/core/v1/core_client.go) and notice how it does things like:

```go
func (c *CoreV1Client) Pods(namespace string) PodInterface {
```

That's what we're going for here, except instead of returning a `BrokerInterface` here, we're returning `BrokerStorage` (we'll change the names later, as I mentioned above).